src/document/docUtil.ts

```diff
@@ -4,6 +4,16 @@ type Selection = {
     range: Range;
 };
 
+declare global {
+    interface Document {
+        documentMode?: number;
+    }
+}
+
+export function isIE(): boolean {
```

Comment:
Can this be avoided by checking for a different property on the selection and/or range(s)?

Reply:
I looked at `Selection.prototype` and `Range.prototype` in IE console and didn't find any similar property, unfortunately

Reply:
Is checking `isCollapsed` and `rangeCount` adequate? Can `rangeCount` be non-zero for a Caret selection?

Reply:
I actually think `isCollapsed` itself is adequate. `rangeCount` is `1` for `Caret` selection.

Reply:
I find `rangeCount` is `0` for `None` type selection. So yeah I think we can remove type check.